### PR TITLE
Fixing return type of Realm.Sync.addListener in API doc and TS definition

### LINKED ISSUE #2566: Wrong typescript/jsdocs for Realm.Sync.addListener
The docs make it seem like it's a void function when in reality, it's a promise.

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #2566 

## ☑️ ToDos

* [x] 📝 Changelog entry
* ~[ ] 📝 `Compatibility` label is updated or copied from previous entry~
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: docs/sync.js [hunk 1]
```diff
@@ -163,6 +163,7 @@ class Sync {
      * @param {string} filterRegex - A regular expression used to determine which changed Realms should trigger events. Use `.*` to match all Realms.
      * @param {string} name - The name of the event.
      * @param {function(changeEvent)} changeCallback - The callback to invoke with the events.
+     * @return {Promise<void>} A promise which is resolved when the worker has started.
```
Comment (blagoev) [inline docs/sync.js]: @returns ?

File: docs/sync.js [hunk 2]
```diff
@@ -192,6 +192,7 @@ class Sync {
      * @param {Realm.Sync.RealmListenerConfiguration} config - The configuration object for Realms being observed.
      * @param {string} eventName - The name of the event to observe.
      * @param {function(changeEvent)} changeCallback - The callback to invoke with the events.
+     * @return {Promise<void>} A promise which is resolved when the worker has started.
```
Comment (blagoev) [inline docs/sync.js]: @returns ?

File: docs/sync.js (hunk 1 above)
Comment (kneth) [reply docs/sync.js]: I think both are allowed.

Comment (blagoev) [reply docs/sync.js]: oh if so proceed at will. maybe for consistency. but no big deal

**Pull request merged**